### ISSUE TITLE
Electron runtime detection

### ISSUE THREAD

**Issue by es128** [mapbox/node-pre-gyp#188]
From: https://github.com/strongloop/fsevents/issues/93#issuecomment-156270459 (//cc @stdavis)

```
node-pre-gyp ERR! install error 
node-pre-gyp ERR! stack Error: Unsupported target version: 0.34.0
node-pre-gyp ERR! stack     at get_runtime_abi (/Users/stdavis/Documents/Projects/atom-amdbutler/node_modules/chokidar/node_modules/fsevents/node_modules/node-pre-gyp/lib/util/versioning.js:156:23)
node-pre-gyp ERR! stack     at Object.module.exports.evaluate (/Users/stdavis/Documents/Projects/atom-amdbutler/node_modules/chokidar/node_modules/fsevents/node_modules/node-pre-gyp/lib/util/versioning.js:265:19)
node-pre-gyp ERR! stack     at install (/Users/stdavis/Documents/Projects/atom-amdbutler/node_modules/chokidar/node_modules/fsevents/node_modules/node-pre-gyp/lib/install.js:138:31)
node-pre-gyp ERR! stack     at Object.self.commands.(anonymous function) [as install] (/Users/stdavis/Documents/Projects/atom-amdbutler/node_modules/chokidar/node_modules/fsevents/node_modules/node-pre-gyp/lib/node-pre-gyp.js:48:37)
node-pre-gyp ERR! stack     at run (/Users/stdavis/Documents/Projects/atom-amdbutler/node_modules/chokidar/node_modules/fsevents/node_modules/node-pre-gyp/bin/node-pre-gyp:79:30)
node-pre-gyp ERR! stack     at Object.&lt;anonymous&gt; (/Users/stdavis/Documents/Projects/atom-amdbutler/node_modules/chokidar/node_modules/fsevents/node_modules/node-pre-gyp/bin/node-pre-gyp:131:1)
node-pre-gyp ERR! stack     at Module._compile (module.js:456:26)
node-pre-gyp ERR! stack     at Object.Module._extensions..js (module.js:474:10)
node-pre-gyp ERR! stack     at Module.load (module.js:356:32)
node-pre-gyp ERR! stack     at Function.Module._load (module.js:312:12)
node-pre-gyp ERR! System Darwin 15.0.0
node-pre-gyp ERR! command "node" "/Users/stdavis/Documents/Projects/atom-amdbutler/node_modules/chokidar/node_modules/fsevents/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Users/stdavis/Documents/Projects/atom-amdbutler/node_modules/chokidar/node_modules/fsevents
node-pre-gyp ERR! node -v v0.10.40
node-pre-gyp ERR! node-pre-gyp -v v0.6.15
node-pre-gyp ERR! not ok 

npm ERR! Darwin 15.0.0
npm ERR! argv "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Users/stdavis/.atom/.apm/.apmrc" "--userconfig" "/Users/stdavis/.atom/.apmrc" "rebuild" "--target=0.34.0" "--arch=x64"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE
npm ERR! fsevents@1.0.5 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the fsevents@1.0.5 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! This is most likely a problem with the fsevents package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls fsevents
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/stdavis/Documents/Projects/atom-amdbutler/npm-debug.log
```

node-pre-gyp does not appear to try to auto-detect electron: https://github.com/mapbox/node-pre-gyp/blob/2e5590ada88d71ec46a51493a9931f8e95cf8b1a/lib/util/versioning.js#L252

When the [install command is baked in](https://github.com/strongloop/fsevents/blob/78a926235773ff8ebc4565bb5a097432f9165c21/package.json#L17) to a package, how would an end-user be able apply the equivalent of `--runtime=electron` in order to prevent their `npm install` from barfing?

**Comment by es128** [comment]
Appears there are two PRs open to address this: #177 & #187

**Comment by stdavis** [comment]
Thanks, @es128!

**Comment by stdavis** [comment]
Any updates on this issue? I can see that the two PR's are still open. Anything that I can do to help get one of them accepted?

**Comment by hedefalk** [comment]
Hi! I'm really struggling with using chokidar -> fsevents -> node-pre-gyp in an Atom plugin.

apm rebuild still fails for me. Repro is as simple as downloading Atom and running apm rebuild on fsevents.

I reported here:
https://github.com/atom/apm/issues/531

but maybe here is a better place to ask? Seems the open issues just lie around…

**Comment by springmeyer** [comment]
We've fix that is now in the released v0.6.26 (https://github.com/mapbox/node-pre-gyp/pull/187)

**Comment by hedefalk** [comment]
@springmeyer I guess you mean 0.6.25? 

I got hopeful here when I saw this released and the following release of fsevents using this (thanks @es128 ). But fsevents still fails with node-pre-gyp with 0.6.25 using apm rebuild:

```
~/d/p/fs-events-apm-rebuild ❯❯❯ rm -rf node_modules
~/d/p/fs-events-apm-rebuild ❯❯❯ npm install

> fsevents@1.0.11 install /Users/viktor/dev/projects/fs-events-apm-rebuild/node_modules/fsevents
> node-pre-gyp install --fallback-to-build

[fsevents] Success: "/Users/viktor/dev/projects/fs-events-apm-rebuild/node_modules/fsevents/lib/binding/Release/node-v46-darwin-x64/fse.node" is installed via remote
fs-events-apm-rebuild@0.0.0 /Users/viktor/dev/projects/fs-events-apm-rebuild
└─┬ fsevents@1.0.11
  ├── nan@2.2.1
  └─┬ node-pre-gyp@0.6.25
  …


~/d/p/fs-events-apm-rebuild ❯❯❯ apm rebuild
Rebuilding modules ✗
node-pre-gyp ERR! install error
node-pre-gyp ERR! stack Error: Unsupported target version: 0.34.5
node-pre-gyp ERR! stack     at get_runtime_abi (/Users/viktor/dev/projects/fs-events-apm-rebuild/node_modules/fsevents/node_modules/node-pre-gyp/lib/util/versioning.js:156:23)
node-pre-gyp ERR! stack     at Object.module.exports.evaluate (/Users/viktor/dev/projects/fs-events-apm-rebuild/node_modules/fsevents/node_modules/node-pre-gyp/lib/util/versioning.js:277:19)
node-pre-gyp ERR! stack     at install (/Users/viktor/dev/projects/fs-events-apm-rebuild/node_modules/fsevents/node_modules/node-pre-gyp/lib/install.js:138:31)
node-pre-gyp ERR! stack     at Object.self.commands.(anonymous function) [as install] (/Users/viktor/dev/projects/fs-events-apm-rebuild/node_modules/fsevents/node_modules/node-pre-gyp/lib/node-pre-gyp.js:48:37)
node-pre-gyp ERR! stack     at run (/Users/viktor/dev/projects/fs-events-apm-rebuild/node_modules/fsevents/node_modules/node-pre-gyp/bin/node-pre-gyp:79:30)
node-pre-gyp ERR! stack     at Object.<anonymous> (/Users/viktor/dev/projects/fs-events-apm-rebuild/node_modules/fsevents/node_modules/node-pre-gyp/bin/node-pre-gyp:131:1)
node-pre-gyp ERR! stack     at Module._compile (module.js:456:26)
node-pre-gyp ERR! stack     at Object.Module._extensions..js (module.js:474:10)
node-pre-gyp ERR! stack     at Module.load (module.js:356:32)
node-pre-gyp ERR! stack     at Function.Module._load (module.js:312:12)
node-pre-gyp ERR! System Darwin 15.3.0
node-pre-gyp ERR! command "node" "/Users/viktor/dev/projects/fs-events-apm-rebuild/node_modules/fsevents/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /Users/viktor/dev/projects/fs-events-apm-rebuild/node_modules/fsevents
node-pre-gyp ERR! node -v v0.10.40
node-pre-gyp ERR! node-pre-gyp -v v0.6.25
node-pre-gyp ERR! not ok

npm ERR! Darwin 15.3.0
npm ERR! argv "/Applications/Atom.app/Contents/Resources/app/apm/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Users/viktor/.atom/.apm/.apmrc" "--userconfig" "/Users/viktor/.atom/.apmrc" "rebuild" "--target=0.34.5" "--arch=x64"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE
npm ERR! fsevents@1.0.11 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the fsevents@1.0.11 install script 'node-pre-gyp install --fallback-to-build'.
npm ERR! This is most likely a problem with the fsevents package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls fsevents
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/viktor/dev/projects/fs-events-apm-rebuild/npm-debug.log
```
